saml-core/src/main/java/org/keycloak/saml/processing/core/util/XMLSignatureUtil.java

```diff
@@ -494,20 +490,16 @@ public static boolean validateSingleNode(Node signatureNode, final KeyLocator lo
 
         logger.trace("Could not validate signature using ds:KeyInfo/ds:KeyName hint.");
 
-        if (locator instanceof Iterable) {
-            Iterable<Key> availableKeys = (Iterable<Key>) locator;
+        logger.trace("Trying hard to validate XML signature using all available keys.");
 
-            logger.trace("Trying hard to validate XML signature using all available keys.");
-
-            for (Key key : availableKeys) {
-                try {
-                    if (validateUsingKeySelector(signatureNode, new KeySelectorPresetKey(key))) {
-                        return true;
-                    }
-                } catch (XMLSignatureException ex) { // pass through MarshalException
-                    logger.debug("Verification failed: " + ex);
-                    logger.trace(ex);
+        for (Key key : locator) {
+            try {
+                if (validateUsingKeySelector(signatureNode, new KeySelectorPresetKey(key))) {
```

Comment:
Not a blocker but do we need `KeySelectorPresetKey` or we can just use `javax.xml.crypto.KeySelector#singletonKeySelector`?

Reply:
No, not needed I think, I just used the same code that was before. The `KeySelectorPresetKey` is the same than the `KeySelector.singletonKeySelector`. Let me replace it.

Reply:
Done!